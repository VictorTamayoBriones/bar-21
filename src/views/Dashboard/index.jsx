import { Navbar } from '../../components/Navbar';
import { Outlet } from 'react-router-dom';

export const Dash = () =>{
    return(
        <>
            <Navbar/>
            <Outlet/>
        </>
    )
}