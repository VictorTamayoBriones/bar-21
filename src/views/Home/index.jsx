
export const Home = () => {
    return(
        <h2>Landing page in progress</h2>
    )
}