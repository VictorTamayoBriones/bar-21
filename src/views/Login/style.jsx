import styled from 'styled-components';
import { theme } from '../../theme';


export const Container = styled.div`
    width: 100%;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
`;

export const LoginContainer = styled.div`
    width: 70%;
    height: 70vh;
    display: flex;
    border: solid 1px ${ theme.moradoClaro };
    border-radius: 8px;
    box-shadow: 0px 0px 9px 2px ${ theme.moradoClaroTransparente };
    overflow: hidden;
    animation: example 8s infinite;

    img{
        width: 50%;
        height: 70vh;
        object-fit: cover;
    }

    form{
        width: 40%;
        margin: 0 auto;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: start;
        padding: 60px 0px;
        gap: 60px;
        h2{
            color: #fff;
            letter-spacing: 1.3px;
            font-size: 30px;
        }
    }

    @media (max-width: 1000px){
        width: 95%;
        display: block;
        position: relative;

        img{
            width: 100%;
            position: relative;
            opacity: .2;
            z-index: 1;
        }
        form{
            width: 80%;
            left: 10%;
            position: absolute;
            top: 0;
            z-index: 2;
        }
    }

    @keyframes example{
        0% { box-shadow: 0px 0px 9px 2px ${ theme.moradoClaroTransparente }; }
        25% { box-shadow: 0px 0px 9px 2px ${ theme.azulTransparente }; }
        50% { box-shadow: 0px 0px 9px 2px ${ theme.moradoClaroTransparente }; }
        75% { box-shadow: 0px 0px 9px 2px ${ theme.azulTransparente }; }
        100% { box-shadow: 0px 0px 9px 2px ${ theme.moradoClaroTransparente }; }
    }
`;