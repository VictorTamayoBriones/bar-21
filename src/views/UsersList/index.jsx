import axios from 'axios'
import styled from 'styled-components';
import { theme } from '../../theme';
import { useNavigate } from 'react-router-dom';
import { Table } from '../../components/Table';
import { ReactComponent as PencilSvg } from '../../images/pencil.svg';
import { ReactComponent as TrashSvg } from '../../images/trash.svg';
import { useState, useEffect } from 'react/cjs/react.development';

export const UsersList=()=>{

    const navigate = useNavigate();

    const[users, setUsers]=useState([]);

    const getUsers = async ()=>{
        const res = await axios.get('http://localhost:4000/api/users');
        setUsers(res.data);
    }

    const handleClickEdit =(id)=>{
        navigate(`/dashboard/usuarios/${id}`);
    }

    const handleClickDelete = async (id)=>{
        await axios.delete(`http://localhost:4000/api/users/${id}`);
        getUsers();
    }

    useEffect(()=>{
        getUsers();
    }, [])

    return(
        <Table>
            <tbody>
                {
                    users.map(({userName, userType, _id})=>{
                        return(
                            <tr key={_id} >
                                <td>{ userName }</td>
                                <td>{ userType }</td>
                                <td>
                                    {
                                        userType === 'empleado' ? <> <EditSvg title='Edit' onClick={()=>handleClickEdit(_id)} /> <DeleteSvg title='Delete' onClick={()=>handleClickDelete(_id)} /> </> : <EditSvg title='Edit' onClick={()=>handleClickEdit(_id)}/>
                                    }
                                </td>
                            </tr>
                        )
                    })
                }
            </tbody>
        </Table>
    )
}

const EditSvg = styled(PencilSvg)`
    width: 22px;
    height: 22px;
    margin: 5px;
    color: #fff;
    cursor: pointer;
    &:hover{
        color: ${ theme.moradoClaro };
    }
`;

const DeleteSvg = styled(TrashSvg)`
    width: 22px;
    height: 22px;
    margin: 5px;
    transform: translateX(20px);
    color: #fff;
    cursor: pointer;
    &:hover{
        color: ${ theme.moradoClaro };
    }
`;