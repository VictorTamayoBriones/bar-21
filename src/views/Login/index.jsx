import { Container, LoginContainer } from "./style"
import Image from '../../images/loginImg.png';
import { Input } from "../../components/Inputs";
import { Button } from "../../components/Buttons";
import { useState } from "react/cjs/react.development";

export const Login = () => {

    const[user, setUser]=useState('');
    const[pass, setPass]=useState('');

    const handleInputChange = ({target:{name, value}})=> {
        if( name === 'user' ){
            setUser(value);
        }else if( name === 'pass' ){
            setPass(value);
        }
    }

    return(
        <Container>
            <LoginContainer>
                <img src={ Image } alt="bar 21 xalpa" title="bar 21 xalpa" />
                    
                <form>
                    <h2>Sig in</h2>
                    <Input onChange={handleInputChange} type="text" placeholder="Usuario" name="user" required />
                    <Input onChange={handleInputChange} type="password" placeholder="ContraseÃ±a" name="pass" required />
                    <Button>Sig in</Button>
                </form>
            </LoginContainer>
        </Container>
    )
}