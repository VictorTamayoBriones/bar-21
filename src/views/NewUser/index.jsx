import { useParams, useNavigate } from 'react-router-dom';
import { Input } from "../../components/Inputs"
import { Button } from '../../components/Buttons';
import { FormUser, DualContainer } from "./style"
import { useEffect, useState } from 'react/cjs/react.development';
import axios from 'axios';

export const NewUser = () =>{
    const params = useParams();
    const navigate = useNavigate();

    const[userName, setUserName]=useState('');
    const[typeSelected, setTypeSelected]=useState('empleado');
    const[email, setEmail]=useState('');
    const[pass, setpass]=useState('');
    const[repeatPass, setRepeatPass]=useState('');
    const[editing, setEditing]=useState(false);
    const[idEditing, setIdEditing]=useState('');

    const handleChange = ({target:{value, name}})=>{
        if( name === 'userName' ){
            setUserName(value);
        }else if( name === 'typeSelected' ){
            setTypeSelected(value);
        }else if( name === 'email' ){
            setEmail(value);
        }else if( name === 'pass' ){
            setpass(value);
        }else if( name === 'repeatPass' ){
            setRepeatPass( value );
        }
    }

    const getData = async ()=>{
        const res = await axios.get(`http://localhost:4000/api/users/${idEditing}`)
        setUserName(res.data.userName);
        setTypeSelected(res.data.userType);
        setEmail(res.data.userEmail);
    }

    useEffect(()=>{

        if( params.id ){
            setEditing(true);
            setIdEditing(params.id);
        }

        if( editing ){
            getData();
        }

    },[editing]);

    const handleSubmit = async (e) =>{
        e.preventDefault();

        if( userName && typeSelected && email && pass && repeatPass ){
            if( pass === repeatPass ){

                const newUser = {
                    userName: userName,
                    userType: typeSelected,
                    userEmail: email,
                    userPass: pass
                }
                
                if( editing ){
                    await axios.put(`http://localhost:4000/api/users/${idEditing}`, newUser);
                }else{
                    await axios.post('http://localhost:4000/api/users', newUser);
                }

                setUserName('');
                setTypeSelected('empleado');
                setEmail('');
                setpass('');
                setRepeatPass('');

                navigate('/dashboard/usuarios/users-list');
            }
        }

    }

    return(
        <FormUser onSubmit={handleSubmit} >
            <DualContainer>
                <Input type='text' placeholder="Full Name" name='userName' value={userName} onChange={handleChange} />

                <select value={typeSelected} onChange={handleChange} name='typeSelected' >
                    <option value="empleado" >Empleado</option>
                    <option value="admin" >Administrador</option>
                </select>
            </DualContainer>

            <Input type="email" placeholder="Email" name='email' value={email} onChange={handleChange} />

            <DualContainer>
                <Input type="password" placeholder="Password" name='pass' value={pass} onChange={handleChange} />

                <Input type="password" placeholder="Repeat Password" name='repeatPass' value={repeatPass} onChange={handleChange} />
            </DualContainer>

            <Button>{ editing ? 'Update' : 'Save' }</Button>
        </FormUser>
    )
}