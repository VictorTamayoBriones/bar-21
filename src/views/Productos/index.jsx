import { Search } from '../../components/Search';
import { ShowProductos } from '../../components/ShowProductos';
import { Container } from './style';

export const Productos = () => {
    return(
        <Container>
            <Search/>
            <ShowProductos/>
        </Container>
    )
}