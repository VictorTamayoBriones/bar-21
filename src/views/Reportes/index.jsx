import { Reporte } from "../../components/Reporte"
import { Container } from './style';
import styled from 'styled-components';
import { theme } from '../../theme';
import { ReactComponent as FileSvg } from "../../images/file-pdf.svg";

export const Reportes = () =>{
    return(
        <Container>
            <Reporte>
                <p>Generar Reporte de ventas</p>
                <p>Mayo - Agosto de 2022</p>
                <PdfFile/>
            </Reporte>
            <Reporte>
                <p>Generar Reporte de ventas</p>
                <p>Mayo - Agosto de 2022</p>
                <PdfFile/>
            </Reporte>
            <Reporte>
                <p>Generar Reporte de ventas</p>
                <p>Mayo - Agosto de 2022</p>
                <PdfFile/>
            </Reporte>
            <Reporte>
                <p>Generar Reporte de ventas</p>
                <p>Mayo - Agosto de 2022</p>
                <PdfFile/>
            </Reporte>
        </Container>
    )
}

const PdfFile = styled(FileSvg)`
    width: 60px;
    height: 60px;
    color: ${ theme.moradoOscuro };
    position: absolute;
    bottom: 12px;
    right: 12px;
    transition: ease all .5s;
`;