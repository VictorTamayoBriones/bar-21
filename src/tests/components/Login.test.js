import React from 'react';
import { shallow } from "enzyme"
import { Login } from "../../views/Login"

describe('Test in <Login/>', () => {
    
    const wrapper = shallow( <Login/> )

    test('should show corectly <Login/>', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('should can change the text box', () => {
        
        const Input = wrapper.find('input');
        console.log(Input);

    })
    

})
