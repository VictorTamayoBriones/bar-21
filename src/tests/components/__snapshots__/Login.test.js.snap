// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Login/> should show corectly <Login/> 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="bar 21 xalpa"
      src="loginImg.png"
      title="bar 21 xalpa"
    />
    <form>
      <h2>
        Sig in
      </h2>
      <styled.input
        name="user"
        onChange={[Function]}
        placeholder="Usuario"
        required={true}
        type="text"
      />
      <styled.input
        name="pass"
        onChange={[Function]}
        placeholder="ContraseÃ±a"
        required={true}
        type="password"
      />
      <styled.button>
        Sig in
      </styled.button>
    </form>
  </styled.div>
</styled.div>
`;
